### PR TITLE
Add WSF-M (Weather Satellite Follow-On - Microwave) soil moisture retrieval and data assimilation support

### PR DESCRIPTION
**Pull request by ejalilva**
## Overview
This PR adds support for WSF-M (Weather Satellite Follow-On - Microwave) soil 
moisture retrieval and data assimilation into LISF.

WSF-M is a passive microwave sensor that measures brightness temperatures across 
multiple channels. Soil moisture retrieval follows a two-step approach:

**Step 1 — Brightness Temperature Resampling (LDT):** Raw WSF-M SDR brightness 
temperatures are resampled to the ARFS grid using an inverse-distance weighting 
approach, with quality filtering for snow, precipitation, and ocean masking, and 
support for separating ascending and descending overpasses.

**Step 2 — ML-Based Soil Moisture Retrieval:** The global land surface is first 
pre-classified into 30 environmental clusters based on land cover, Köppen climate 
zone, and soil texture. A dedicated Random Forest model is then trained for each 
cluster, yielding 30 cluster-specific models plus one global fallback model for 
unclassified pixels. A Gaussian Mixture Model (GMM) is used for soft cluster 
assignment, blending predictions from neighboring clusters at boundaries to 
eliminate hard transition artifacts. Together, these components form a spatially 
adaptive retrieval system that translates resampled brightness temperatures into 
volumetric soil moisture estimates.

The retrieved soil moisture is integrated into the LIS data assimilation framework, 
supporting CDF-based bias correction and assimilation into Noah.3.9 and Noah-MP.4.0.1 
land surface models. Both single-variate (WSF-M only) and multi-variate 
(WSF-M + SMOPS + SMAP) configurations are supported.

### LDT Changes
- Added `ldt/WSF_OPL/` — core WSF-M brightness temperature resampling module
- Added `ldt/DAobs/WSF_sm/` — WSF-M soil moisture DA observation preprocessing for CDF generation
- Added `ldt/runmodes/WSF_OPL/` — LDT run mode registration for WSF-M OPL
- Updated `ldt/plugins/LDT_pluginIndices.F90` — registered WSF-M OPL plugin index
- Updated `ldt/plugins/LDT_runmode_pluginMod.F90` — registered WSF-M OPL init/run routines
- Updated `ldt/make/Filepath` — added WSF_OPL and DAobs/WSF_sm paths
- Added `ldt/WSF_OPL/prediction/predict_sm_stratified.py` — ML soil moisture prediction script

### LIS Changes
- Added `lis/dataassim/obs/WSF_sm/` — WSF-M soil moisture observation reader for DA
- Updated LIS plugin indices and DA obs plugin to register WSF-M sm reader
- Updated `lis/make/default.cfg` — added WSF-M configuration entry

## Test Cases
All test cases are available on Discover at:

**WSF-M Resampling & Soil Moisture Retrieval:**
`/discover/nobackup/projects/usaf_lis/ejalilvand/Test_cases/wsf_integerated/wsf_sm_retrieval`
A README file is included explaining the WSF-M SM retrieval workflow.

**CDF Generation:**
- Global: `/discover/nobackup/projects/usaf_lis/ejalilvand/Test_cases/wsf_integerated/wsf_cdf_generation`
- CONUS: `/discover/nobackup/projects/usaf_lis/ejalilvand/Test_cases/wsf_integerated/wsf_cdf_generation/cdf_conus`
- Spatial sampling: WSF cdf created with the spatial sampling option will be added

**Data Assimilation — Global Functional Tests:**
`/discover/nobackup/projects/usaf_lis/ejalilvand/Test_cases/wsf_integerated/wsf_sm_da`
Includes test cases for:
- Noah.3.9 and Noah-MP.4.0.1
- Single-variate WSF-M soil moisture DA
- Multi-variate DA (WSF-M + SMOPS + SMAP)
- Data denial experiments

**Data Assimilation — CONUS Tests:**
- Single-variate (Noah.3.9 and Noah-MP.4.0.1): `/discover/nobackup/projects/usaf_lis/ejalilvand/Test_cases/wsf_integerated/CONUS/da_conus`
- Multi-variate (Noah.3.9 and Noah-MP.4.0.1): `/discover/nobackup/projects/usaf_lis/ejalilvand/Test_cases/wsf_integerated/CONUS/da_conus_mv`

### ML Model Files
The following pre-trained model files are required and available on Discover 
alongside the test case:
- 30 per-cluster Random Forest models (`rf_cluster_00.pkl` – `rf_cluster_29.pkl`)
- 1 global fallback model (`rf_global.pkl`)
- 1 GMM soft-assignment model (`gmm_k30.pkl`)
- 1 KMeans clustering model (`clustering_model_k30.pkl`)
- 1 cluster map (`environmental_clusters_k30.nc`)
- 1 configuration file (`stratified_config.json`)

### Environmental Data Layers
- Land cover, Köppen climate zone, and soil texture (clay/sand/silt) — 
paths configured via `pipeline_config.json` in the test case directory.

### Python Environment
A conda `environment.yml` and detailed `README.md` are provided in the 
retrieval test case directory:
`/discover/nobackup/projects/usaf_lis/ejalilvand/Test_cases/wsf_integerated/wsf_sm_retrieval`

Key requirement: **scikit-learn 1.7.1** (must match the training version — 
using a different version will produce incorrect predictions).
```bash
conda env create -f environment.yml
conda activate wsf_sm
```

### CODE REVIEW

**Comment by jvgeiger** [comment]
This pull request will be closed.  See PR #1799.
